zerver/views/users.py

```diff
@@ -134,6 +137,8 @@ def deactivate_user_backend(
         request,
         user_profile,
         target,
+        is_spammer=is_spammer,
+        message_delete_action=message_delete_action,
```

Comment:
Somewhere we need to be sure that the organization's `can_delete_message_group` policy is checked for the acting user.